Xamarin.Forms.Platform.Android/AppCompat/TabbedPageRenderer.cs

```diff
@@ -10,22 +10,39 @@
 using Android.Support.Design.Widget;
 using Android.Support.V4.App;
 using Android.Support.V4.View;
+using AWidget = Android.Widget;
 using Android.Views;
 using Xamarin.Forms.Internals;
 using Xamarin.Forms.PlatformConfiguration.AndroidSpecific;
+using ADrawableCompat = Android.Support.V4.Graphics.Drawable.DrawableCompat;
+using AColor = Android.Graphics.Color;
+using AView = Android.Views.View;
+using AMenu = Android.Views.Menu;
 
 namespace Xamarin.Forms.Platform.Android.AppCompat
 {
-	public class TabbedPageRenderer : VisualElementRenderer<TabbedPage>, TabLayout.IOnTabSelectedListener, ViewPager.IOnPageChangeListener, IManageFragments
+	public class TabbedPageRenderer : VisualElementRenderer<TabbedPage>, TabLayout.IOnTabSelectedListener, ViewPager.IOnPageChangeListener, IManageFragments, BottomNavigationView.IOnNavigationItemSelectedListener
 	{
 		Drawable _backgroundDrawable;
-		int? _defaultColor;
+		ColorStateList _originalTabTextColors;
+		ColorStateList _orignalTabIconColors;
+
+		ColorStateList _newTabTextColors;
+		ColorStateList _newTabIconColors;
+
 		bool _disposed;
 		FragmentManager _fragmentManager;
 		TabLayout _tabLayout;
+		BottomNavigationView _bottomNavigationView;
+		AWidget.RelativeLayout _relativeLayout;
 		bool _useAnimations = true;
 		FormsViewPager _viewPager;
 		Page _previousPage;
+		int[] checkedStateSet = null;
+		int[] selectedStateSet = null;
+		int[] emptyStateSet = null;
+		int DefaultARGBColor = Color.Default.ToAndroid().ToArgb();
+		AColor DefaultAndroidColor = Color.Default.ToAndroid();
```

Comment:
Shouldn't we grab the default color from Android and not forms?

Reply:
The point of these aren't to set the color to anything really. It's just to check if I should remove any colors I've applied and then just let Android do its natural thing. I use them for the Drawable tints to see if I should just remove the tint and let Android do its thing

```c#
if (colors.GetColorForState(GetSelectedStateSet(), _defaultAndroidColor) == _defaultARGBColor)
     ADrawableCompat.SetTintList(icon, null);
```